guides/v2.1/cloud/release-notes/cloud-tools.md

```diff
@@ -25,17 +25,60 @@ php ./vendor/bin/ece-tools list
+
+- <!--MAGECLOUD-2517-->**Deployment improvement**—Added validation to warn customers if a Staging or Production environment has `dev`, `debug`, or `debug_logging` options enabled.
+
+
+#### Resolved Issues
+
+-  **Deployment fixes**
+	
+    -  <!--MAGECLOUD-2603-->Now maintenance mode is enabled at the start of the deploy phase and disabled at the end. If the deployment fails, the site remains in maintenance mode until the deployment issues are resolved.
+
+    -  Reworked the deploy phase validation checks to downgrade the error level for the following deployment issues from `CRITICAL` to `WARNING` so that the deployment completes. Previously, these issues caused the deployment to fail.
+
+       -  ADMIN_EMAIL is not set on upgrade.
+	   
+	   -  ADMIN_EMAIL or ADMIN_USERNAME is associated with another account.
+
+       -  Environment configuration contains incorrect values for deploy or cloud variables.
+
+       -  <!--MAGECLOUD-2600-->The Elasticsearch version on the cloud infrastructure is incompatible with the version of the elasticsearch/elasticsearch module supported by {{site.data.var.ece}}. See the [Elasticsearch troubleshooting article](https://support.magento.com/hc/en-us/articles/360015758471-Deployment-fails-or-interrupts-with-cloud-log-error-Elasticsearch-version-is-not-compatible-with-current-version-of-magento) in the Magento Support Knowledgebase.
+
+    - <!--MAGECLOUD-2173-->Fixed an issue with the shared configuration settings in the `app/etc/config.php` file that caused `recursion detected` errors during deployment.
+
+-  **Cron-related fixes**
+
+   -  <!--MAGECLOUD-2602-->Fixed a cron scheduling issue that prevented jobs from running if you specify a cron frequency other than the default (1 minute).
+
+   -  <!--MAGECLOUD--2537-->Fixed an issue in the deploy phase that allowed cron jobs to continue running during deployment, which can cause database locks and other critical issues. Now, all cron jobs are stopped before the deploy phase begins and restarted after the deployment completes.
+   
+   -  <!--MAGECLOUD-2501-->Fixed the cron job workflow in versions 2.1.4 through 2.1.x to unlock frozen cron jobs so that they can be stopped before beginning deployment. Previously, a frozen cron job caused the deployment to stall.
+   
+-  <!--MAGECLOUD-2527-->Changed the format of the `config.php` file generated by the `vendor/bin/ece-tools config:dump` command to use short array syntax and 4-space indentation to comply with Magento coding standards./:q
```

Comment:
Looks like there are extra random characters at the end of this line?

>... coding standards./:q

Reply:
Maybe it's an emoji. 😯